tests/build_/test_plugins_configuration.py

```diff
@@ -294,9 +294,12 @@ def test_render_flatpak(self, compose_ids, build_type):
             assert get_plugin(plugins, "postbuild_plugins", "import_image")
 
     @pytest.mark.parametrize('build_type', (BUILD_TYPE_ORCHESTRATOR, BUILD_TYPE_WORKER))
-    def test_render_prod_not_flatpak(self, build_type):
+    @pytest.mark.parametrize('flatpak', (False, True))
+    @pytest.mark.parametrize('flatpak_base_image', (TEST_FLATPAK_BASE_IMAGE, False))
+    def test_render_prod_not_flatpak(self, build_type, flatpak, flatpak_base_image):
```

Comment:
It doesn't make sense to me to add flatpak True/False to a test that is called _not_flatpak - the point of this test was to check what happens when Flatpak is not set. Parameterization here of flatpak_base_image here would make sense - and should have no effect on the resulting set of plugins. If you wanted to check what happens for flatpak = True, flatpak_base_image = False then that would belong in test_render_flatpak.

Right now, I think you get a plugin configuration that will make atomic-reactor fail -with resolve_module_compose and flatpak_create_oci plugins, but no flatpak_create_dockerfile plugin. You could assert that you get that, but it would certainly be better to fail in osbs-client with an informative error message about the missing config key, rather than fail in atomic-reactor with '"flatpak_create_dockerfile must be run before flatpak_create_oci" which doesn't help the poor admin at all!

[ As far as I can see, flatpak_base_image should have moved to the reactor config and out of osbs.conf with other site-static configuration. It was probably left over because it sounds like a "base image" - that you might want different for different builds. But actually, the base image should have little on the final contents of the generated Flatpak OCI - it's just the image that provides rpm/dnf/etc. to assemble the contents. So we should fix that up at some point, but let's not do it in this PR. There's not much harm in having it left over since we're not doing openshift-triggered layered image rebuilds for flatpaks. ]

Reply:
Filed https://github.com/projectatomic/osbs-client/issues/784 for tracking `flatpak_base_image` move to reactor config map.